_posts/2019-05-23-assessing-rails-upgrades.markdown

```diff
@@ -0,0 +1,32 @@
+---
+layout: post
+title: "Is this Rails Upgrade Project a good fit?"
+date: 2019-05-23 15:00:00
+categories: ["rails", "upgrades"]
+author: "emily"
+---
+
+Over the years we have seen a significant increase in the  demand for [Ruby on Rails](http://rubyonrails.org/) upgrades. That's why a few years ago we decided to launch [FastRuby.io](https://fastruby.io), our Ruby on Rails upgrade service.
+
+When we receive an inquiry via [FastRuby.io](https://fastruby.io), **the first thing we like to do is schedule a call with the potential client to get more details about their app**.
+
+<!--more-->
+
+In the call we like to get the following info:
+
+1. A general overview of their app and why they are interested in upgrading Rails
+2. What **Rails version is their app currently on**?
+3. What **Rails version do they want to upgrade to**?
+4. What **Ruby version** is their app using?
+5. How large is their app? How many models does it have?
+6. **How much test coverage does their app have**?
+7. Are there **any major blockers** that they know of? Have they tried to upgrade before?
+8. Do they have any **time limitations or deadlines**? How fast do they want to start working on the upgrade?
+
+These questions help us get an idea of the scope and level of risk of the Rails upgrade. Asking the potential client about the test coverage on their app is crucial to assess risk. **We generally don’t upgrade Rails apps that have less than 60% test coverage**. Similarly, asking about the potential client’s timeframe and deadlines can be helpful to determine if we have the availability and bandwidth to complete the Rails upgrade. Depending on the urgency of the project, we might assign two developers to work full time or part time on the project.
+
+Based on the information from these initial [FastRuby.io](https://fastruby.io) calls, we can determine whether or not the Rails upgrade project is a good fit for the team.
+
+If the project is a good fit for us, we direct our leads to purchase [The Roadmap](https://fastruby.io/roadmap). All of our clients purchase this initial assessment to get an idea of the state of their project. After they receive the Roadmap Report, some clients decide to use the roadmap to work on a Rails upgrade on their own and others decide to work with us on their upgrade project.
```

Comment:
@ercohen14 Capitalization is a little weird with Roadmap Report, maybe use "The Roadmap's report"

Reply:
Also, could you link Rails upgrade to https://fastruby.io/blog/tags/upgrades?